### PR TITLE
[SHACK-354] Use chef update API  to check for chef-workstation product updates.

### PR DESCRIPTION
**Pull request by marcparadise**
This does the following: 
- update checks are re-enabled
- the update dialog now shows the correct installed WS version (was showing  app version)
- updates hit against omnitruck
- we use the chef-workstation ohai to gather platform data to use in querying omnitruck.
- clicking 'download' will take you to the package url
- added some error handling when platform is unsupported (the default case on my arch box...)
- we will now not show the notification icon in tray when you're explicitly requesting update check, because we're popping up a notification dialog.

In order to keep the UI bits separated from the backend-behaviors like
update checking, we're emitting update-related events to be handled by
the UI , instead of managing UI elements directly from the updater. 

Signed-off-by: Marc A. Paradise <marc.paradise@gmail.com>

### CODE REVIEW

**Comment by marcparadise** [comment]
I've moved the update checks to occur back on the main thread (for now) because with the event/listener model , all actions so far seem to be non-blocking. Will be reading and testing further tomorrow to make sure.

File: main.js [hunk 1]
```diff
@@ -75,24 +76,83 @@ function createTray() {
 }
 
 function startApp() {
-  const modalPath = `file://${__dirname}/process.html`
+  const modalPath = `file://${__dirname}/process.html` // TODO about:blank?
```
Comment (tyler-ball) [inline main.js]: In order to get the tests working, I had to load the `process.js` thread from this html file, even though the window is not visible. Happy to look into doing it another way - we can refactor and as long as the tests continue running we should be good!

File: main.js [hunk 2]
```diff
@@ -75,24 +76,83 @@ function createTray() {
...
+    noUpdateDialog.open();
+  }
+  WSTray.instance().displayNotification(false);
+});
+
+mixlibInstallUpdater.on('update-available', (updateInfo) => {
+  console.log("got update-available: " + util.inspect(updateInfo));
+  // Only display the notification. Changing the menu text is a lot of work
+  // and will be done in the next re-factor.
+  //  TODO - why ? This ... *looked* like we could just update the label...
```
Comment (tyler-ball) [inline main.js]: https://github.com/hovancik/stretchly/blob/master/app/main.js#L510

One of the other tray app / windows based applications I have looked at uses a `getMenuTray()` method and calls `setContextMenu` any time they need to update the menu. I think Jon's idea about pushing this logic into a class that can manage existing state and be responsible for the updates makes sense.

File: main.js [hunk 3]
```diff
@@ -40,14 +43,12 @@ function createMenu() {
...
-    //
-    // {
-    //   label: 'Check for updates...',
-    //   click: updater.checkForUpdates
-    // },
-    // {type: 'separator'},
+    {
+      id: 'updateCheck',
+      label: 'Check for updates...',
+      click: () => { requestFromMenu = true;  app.emit('do-update-check') }
```
Comment (tyler-ball) [inline main.js]: I understand why you have `requestFromMenu` stored as state on the main process. It originally seemed weird to me but now I think it makes sense - the main process is responsible for 'drawing' our tray app and the logic for whether this request came from the menu only lives here. In this sense, `requestFromMenu` should absolutely be state we store in the main process.

I had to look at this for a long time before understanding this was correct. I think I still like the idea that the main process sets the initial application state and then starts a renderer process that is responsible for all the 'drawing' related functions of our tray application. But thats a refactor we can do later.

Comment (marcparadise) [reply main.js]: Yep, I like that idea too - for this pass I just wanted to get us switched over.  We can talk about restructuring how we handle render/main process.

**Comment by tyler-ball** [comment]
Referencing https://github.com/chef/chef-workstation/pull/249#issuecomment-426065206 because this PR may be a good time to address that

**Comment by tyler-ball** [comment]
@marcparadise rebased and force pushed

**Pull request merged**